clients/spark/core/src/main/resources/catalog.proto

```diff
@@ -12,4 +12,10 @@ message Entry {
 	int64 size = 3;
 	string e_tag = 4;
 	map<string,string> metadata = 5;
+	enum AddressType {
+		UNKNOWN = 0;
```

Comment:
As an alternative, we might:
1. Rename the field `address` -> `deprecated_by_prefix_address` (or otherwise discourage its use).
1. Add a new field `string address = 7`.
1. Have just `RELATIVE` (0, default) and `FULL` address types.
1. Document that `address_type` applies _only_ to `address`.

This ensures we can distinguish missing type fields to see whether they are old (and intentional) or new (and therefore an error).